### PR TITLE
Fix issue with env variables not being inserted in forward model jobs

### PR DESCRIPTION
**Pull request by eivindjahren**
Fix an issue where environment variables did not enter the forward model.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Updated documentation
- [x] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/clib/lib/config/config_parser.cpp [hunk 1]
```diff
@@ -565,31 +574,54 @@ config_parse__(config_parser_type *config, config_content_type *content,
...
-                    token_list, 2, active_tokens, " ");
+                if (active_tokens < 3) {
+                    content->parse_errors.push_back(fmt::format(
+                        "Keyword:{} must have two or more arguments.",
+                        define_kw));
+                    return;
+                }
 
-                config_content_add_define(content, key, value);
+                const std::string key = util_alloc_string_copy(
```
Comment (andreas-el) [inline src/clib/lib/config/config_parser.cpp:584]: Is it necessary to util_allow_string_copy this? Also, should you free the char* string if so?

File: src/clib/lib/config/config_parser.cpp [hunk 2]
```diff
@@ -565,31 +574,54 @@ config_parse__(config_parser_type *config, config_content_type *content,
...
-                               "arguments. \n",
-                               __func__, define_kw);
-
-                char *key = (char *)util_alloc_string_copy(
-                    stringlist_iget(token_list, 1));
-                char *value = stringlist_alloc_joined_substring(
-                    token_list, 2, active_tokens, " ");
+                if (active_tokens < 3) {
+                    content->parse_errors.push_back(fmt::format(
+                        "Keyword:{} must have two or more arguments.",
```
Comment (andreas-el) [inline src/clib/lib/config/config_parser.cpp:579]: Feedback is now correct? I take it that active_tokens 3 implies two arguments?

Comment (eivindjahren) [reply src/clib/lib/config/config_parser.cpp:579]: Yes :)

File: src/clib/lib/config/config_parser.cpp [hunk 3]
```diff
@@ -565,31 +576,53 @@ config_parse__(config_parser_type *config, config_content_type *content,
...
+                            char_to_optstr(res_env_isscanf_alloc_envvar(
+                                stringlist_iget(token_list.get(), iarg),
+                                env_offset));
+                        if (!env_var.has_value())
+                            break;
+
+                        const char *env_value =
+                            getenv(env_var.value().substr(1).c_str());
+                        if (env_value != nullptr) {
+                            char *new_value = util_string_replace_alloc(
```
Comment (andreas-el) [inline src/clib/lib/config/config_parser.cpp:601]: This one will need to be freed also? Do you even need to allocate for this?

File: src/clib/lib/config/config_parser.cpp [hunk 4]
```diff
@@ -565,31 +576,53 @@ config_parse__(config_parser_type *config, config_content_type *content,
...
+                                                      new_value);
+                        } else {
+                            env_offset += 1;
+                            logger->warning(
+                                "Environment variable: {} is not defined",
+                                env_var.value());
+                        }
+                    }
+                }
+                const std::string value = stringlist_alloc_joined_substring(
```
Comment (andreas-el) [inline src/clib/lib/config/config_parser.cpp:614]: Do we need allocation for this?

File: src/clib/lib/config/config_parser.cpp [hunk 5]
```diff
@@ -435,6 +436,12 @@ bool config_parser_add_key_values(
     return new_node != NULL;
 }
 
+static std::optional<std::string> char_to_optstr(char *p) {
```
Comment (andreas-el) [inline src/clib/lib/config/config_parser.cpp:439]: As a user of this function, i'd expect it not to free my string when converting?

File: src/clib/lib/config/config_parser.cpp (hunk 1 above)
Comment (eivindjahren) [reply src/clib/lib/config/config_parser.cpp]: Ah, no, not any more as assigning `char*` to `std::string` is a copy action. Fixed it :)

File: src/clib/lib/config/config_parser.cpp (hunk 3 above)
Comment (eivindjahren) [reply src/clib/lib/config/config_parser.cpp]: Fixed this with use of unique_ptr

File: src/clib/lib/config/config_parser.cpp (hunk 4 above)
Comment (eivindjahren) [reply src/clib/lib/config/config_parser.cpp]: Fixed it with use of unique_ptr

File: src/clib/lib/config/config_parser.cpp (hunk 5 above)
Comment (eivindjahren) [reply src/clib/lib/config/config_parser.cpp]: Yea, I changed it to take unique_ptr as pass by value so that the expectation is that it takes ownership.

**Review by andreas-el — APPROVED** [review]
As far as i can see, this seems okay. :+1:

**Pull request merged**